### PR TITLE
Add account address form tests

### PR DESCRIPTION
**Pull request by AntonoffEvgeniy**
Add account address form tests

### CODE REVIEW

File: ShopAppTests/App/Account/Controllers/AccountAddressFormViewControllerSpec.swift [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                viewController.selectedAddress = Address()
+                _ = viewController.view
+            }
+            
+            it("should have a correct view model type") {
+                expect(viewController.viewModel).to(beAKindOf(AccountAddressFormViewModel.self))
+            }
+            
+            it("should have a back button") {
+                expect(viewController.navigationItem.leftBarButtonItem?.image) == #imageLiteral(resourceName: "arrow_left")
```
Comment (plzen) [inline ShopAppTests/App/Account/Controllers/AccountAddressFormViewControllerSpec.swift]: I can't find where this item is set in view controller

File: ShopAppTests/App/Account/Controllers/AccountAddressFormViewControllerSpec.swift [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                    expect(childController?.address) === viewController.selectedAddress
+                }
+            }
+        }
+        
+        describe("when address added") {
+            beforeEach {
+                viewController.addressAction = .add
+                _ = viewController.view
+                viewController.viewModel.addCustomerAddress(with: Address())
```
Comment (plzen) [inline ShopAppTests/App/Account/Controllers/AccountAddressFormViewControllerSpec.swift]: Don't call viewModel directly - use delegate for this action

File: ShopAppTests/App/Account/Controllers/AccountAddressFormViewControllerSpec.swift [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            it("should add address") {
+                expect(viewControllerDelegateMock.addressDidAdd) == true
+            }
+        }
+        
+        describe("when address updated") {
+            beforeEach {
+                viewController.addressAction = .edit
+                _ = viewController.view
+                viewController.viewModel.updateCustomerAddress(with: Address())
```
Comment (plzen) [inline ShopAppTests/App/Account/Controllers/AccountAddressFormViewControllerSpec.swift]: Don't call viewModel directly - use delegate for this action

File: ShopAppTests/App/Account/Controllers/AccountAddressFormViewControllerSpec.swift [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+//
+
+import Nimble
+import Quick
+
+import ShopApp_Gateway
+
+@testable import ShopApp
+
+class AccountAddressFormViewControllerSpec: QuickSpec {
```
Comment (plzen) [inline ShopAppTests/App/Account/Controllers/AccountAddressFormViewControllerSpec.swift]: You miss test for viewModel.filledAddress

File: ShopAppTests/App/Account/Controllers/OrderDetailsViewControllerSpec.swift [hunk 5]
```diff
@@ -52,7 +52,7 @@ class OrderDetailsViewControllerSpec: QuickSpec {
             }
             
             it("should have correct view model properties") {
-                expect(viewModel.orderId) == viewController.orderId
+                expect(viewModelMock.orderId) == viewController.orderId
             }
             
             it("should have correct content inset of table view") {
```
Comment (plzen) [inline ShopAppTests/App/Account/Controllers/OrderDetailsViewControllerSpec.swift]: Please, add test for checking that `viewDidLoad` calls `viewModel.loadOrder()`

Comment (plzen) [reply ShopAppTests/App/Account/Controllers/OrderDetailsViewControllerSpec.swift]: Can we add test for backButtonDidPress?

File: ShopAppTests/App/Account/Controllers/OrderListViewControllerSpec.swift [hunk 6]
```diff
@@ -68,6 +68,9 @@ class OrderListViewControllerSpec: QuickSpec {
             
             context("if data is empty") {
```
Comment (plzen) [inline ShopAppTests/App/Account/Controllers/OrderListViewControllerSpec.swift]: Verify that `viewDidLoad` calls `viewModel.reloadData()`

File: ShopAppTests/App/Account/Controllers/OrderListViewControllerSpec.swift [hunk 7]
```diff
@@ -68,6 +68,9 @@ class OrderListViewControllerSpec: QuickSpec {
...
                 it("should have tableView without data") {
```
Comment (plzen) [inline ShopAppTests/App/Account/Controllers/OrderListViewControllerSpec.swift]: Verify `pullToRefreshHandler` method

Comment (plzen) [reply ShopAppTests/App/Account/Controllers/OrderListViewControllerSpec.swift]: Verify `infinityScrollHandler`

File: ShopAppTests/App/Account/ViewModels/AccountAddressFormViewModelSpec.swift [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+            var address: Address!
+            
+            beforeEach {
+                disposeBag = DisposeBag()
+                address = Address()
+                address.id = "address id"
+            }
+            
+            context("if address added") {
+                it("should add address") {
```
Comment (plzen) [inline ShopAppTests/App/Account/ViewModels/AccountAddressFormViewModelSpec.swift]: Verify error state

File: ShopAppTests/App/Account/ViewModels/AccountAddressFormViewModelSpec.swift [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+                    viewModel.filledAddress
+                        .subscribe(onNext: { result in
+                            expect(result) === address
+                        })
+                        .disposed(by: disposeBag)
+                }
+            }
+            
+            context("if address updated") {
+                it("should update address") {
```
Comment (plzen) [inline ShopAppTests/App/Account/ViewModels/AccountAddressFormViewModelSpec.swift]: Verify error state

File: ShopAppTests/App/Account/ViewModels/AccountAddressFormViewModelSpec.swift (hunk 8 above)
Comment (plzen) [reply ShopAppTests/App/Account/ViewModels/AccountAddressFormViewModelSpec.swift]: Verify changing state of view

File: ShopAppTests/App/Account/ViewModels/AccountAddressFormViewModelSpec.swift (hunk 9 above)
Comment (plzen) [reply ShopAppTests/App/Account/ViewModels/AccountAddressFormViewModelSpec.swift]: Verify changing state of view

File: ShopAppTests/App/Account/Controllers/AccountAddressFormViewControllerSpec.swift (hunk 1 above)
Comment (AntonoffEvgeniy) [reply ShopAppTests/App/Account/Controllers/AccountAddressFormViewControllerSpec.swift]: In BaseViewController

File: ShopAppTests/App/Account/Controllers/AccountAddressFormViewControllerSpec.swift [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+            beforeEach {
+                disposeBag = DisposeBag()
+                address = Address()
+            }
+            
+            it("should return correct address") {
+                viewModelMock.filledAddress.onNext(address)
+                viewController.viewModel.filledAddress
+                    .subscribe(onNext: { filledAddress in
+                        expect(filledAddress) === address
```
Comment (plzen) [inline ShopAppTests/App/Account/Controllers/AccountAddressFormViewControllerSpec.swift]: This test doesn't cover logic from AccountAddressFormViewController class. You need to check that emission of filledAddress calls delegate with returned value

**Pull request merged**